test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt

```diff
@@ -231,6 +232,40 @@ class SyncedRealmTests {
         realm3.close()
     }
 
+    @Test
+    fun errorHandlerReceivesPermissionDeniedSyncError() {
+        val channel = Channel<SyncException>(1).freeze()
+        val (email, password) = randomEmail() to "password1234"
+        val user = runBlocking {
+            app.createUserAndLogIn(email, password)
+        }
+
+        val config = SyncConfiguration.Builder(
+            schema = setOf(ParentPk::class, ChildPk::class),
+            user = user,
+            partitionValue = partitionValue
+        ).errorHandler { _, error ->
+            channel.trySend(error)
+        }.build()
+
+        runBlocking {
+            // Remove permissions to generate a sync error containing ONLY the original path
+            // This way we assert we don't read wrong data from the user_info field
+            app.asTestApp.changeSyncPermissions(SyncPermissions(read = false, write = false))
```

Comment:
We are changing the server config but there is no guarantee to return it back to the original state if the test fails. To prevent cascading issues to other tests it can be wrapped with a finally block.

Another thing is that we could encapsulate this logic on a `testApp.generateSyncError(block: ()->Unit)` that automatically does the permissions change, exception catching and restore. That way we separate the logic of generating the exception from the actual test case.